From 735f8e96bfd7047c13f5f0edb351ff944ce6b6ff Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Wed, 12 Sep 2012 13:10:22 -0300
Subject: [RHEL6 qemu-kvm PATCH 03/14] usb-redir: Reset device address and
 speed on disconnect

RH-Author: Hans de Goede <hdegoede@redhat.com>
Message-id: <1347455431-11260-4-git-send-email-hdegoede@redhat.com>
Patchwork-id: 41848
O-Subject: [RHEL-6.4 qemu-kvm PATCH 03/12] usb-redir: Reset device address and speed on disconnect
Bugzilla:
RH-Acked-by: Alon Levy <alevy@redhat.com>
RH-Acked-by: Gerd Hoffmann <kraxel@redhat.com>
RH-Acked-by: Christophe Fergeau <cfergeau@redhat.com>
Bugzilla-related: 805172

Without this disconnected devices look like the last redirected device
in the monitor in "info usb".

Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Signed-off-by: Gerd Hoffmann <kraxel@redhat.com>
Upstream commit: a0625c56e0db19029cb69e28859af56b2ae19f3f
Conflicts: usb-redir.c
---
 usb-redir.c | 2 ++
 1 file changed, 2 insertions(+)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 usb-redir.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/usb-redir.c b/usb-redir.c
index 80f67f4..a3f950a 100644
--- a/usb-redir.c
+++ b/usb-redir.c
@@ -1156,6 +1156,8 @@ static void usbredir_device_disconnect(void *priv)
         QTAILQ_INIT(&dev->endpoint[i].bufpq);
     }
     dev->interface_info.interface_count = NO_INTERFACE_INFO;
+    dev->dev.addr = 0;
+    dev->dev.speed = 0;
 }
 
 static void usbredir_interface_info(void *priv,
-- 
1.7.11.4

